pymc4/distributions/continuous.py

```diff
@@ -1174,6 +1177,97 @@ def upper_limit(self):
         return self.conditions["high"]
 
 
+class Flat(ContinuousDistribution):
+    r"""A uniform distribution with support :math:`(-\inf, \inf)`.
+    Used as a uninformative log-likelihood that returns
+    zeros regardless of the passed values.
+    """
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name, **kwargs)
+
+    @staticmethod
+    def _init_distribution(conditions):
+        return tfd.Uniform(low=-np.inf, high=np.inf)
+
+    def log_prob(self, value):
+        # convert the value to tensor
+        value = tf.convert_to_tensor(value)
+        expected = tf.zeros(self.batch_shape + self.event_shape)
+        # check if the event shape matches
+        if len(self.event_shape) and value.shape[-len(self.event_shape) :] != self.event_shape:
+            raise ValueError("values not consistent with the event shape of distribution")
+        # broadcast expected to shape of value
+        if len(value.shape) < len(self.batch_shape + self.event_shape):
+            if (
+                value.shape[: -len(self.event_shape)]
```

Comment:
This won't work if the event shape is a scalar, you'll end up with `value.shape[:0]`. You will have to change this to `value.shape[:len(value.shape) - len(event_shape)]`